hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -573,16 +573,19 @@ private ByteString readStateMachineData(
         ReadChunkRequestProto.newBuilder()
             .setBlockID(writeChunkRequestProto.getBlockID())
             .setChunkData(chunkInfo);
-    ContainerCommandRequestProto dataContainerCommandProto =
+    ContainerCommandRequestProto.Builder dataContainerCommandProtoBuilder =
         ContainerCommandRequestProto.newBuilder(requestProto)
-            .setCmdType(Type.ReadChunk).setReadChunk(readChunkRequestProto)
-            .build();
+            .setCmdType(Type.ReadChunk).setReadChunk(readChunkRequestProto);
+    if (requestProto.hasEncodedToken()) {
+      dataContainerCommandProtoBuilder.setEncodedToken(
```

Comment:
I think `newBuilder(requestProto)` sets all fields from `requestProto`, including encoded token, so this seems unnecessary.  Am I missing something?

Reply:
You are right, the constructor does copy the token if exist. I will close this one.